components/MapComponent.tsx

```diff
@@ -1,123 +1,146 @@
-import React, { useState } from "react";
-import { Button, StyleProp, View, ViewStyle } from "react-native";
-import MapView, { Marker, PROVIDER_GOOGLE } from "react-native-maps";
+import { Location } from "lib/location/Location"
+import React, { useState } from "react"
+import { StyleProp, View, ViewStyle } from "react-native"
+import MapView, { Marker, PROVIDER_GOOGLE } from "react-native-maps"
+
+// Type for the map markers, in order to separate their logic out
+export interface mapMarker {
+  key: string
+  title: string
+  location: Location
+}
+
+interface Props<T> {
```

Comment:
This could have a more descriptive name, maybe something like `MapProps` or `MapComponentProps`?